scripts/jirautils.py

```diff
@@ -2,10 +2,77 @@
 
 # This script contains jira related utilties for Jenkins.
 
-
+from collections import defaultdict
+import logging
+import re
 import click
 from jira import JIRA, JIRAError
 
+logging.basicConfig()
+LOGGER = logging.getLogger("jirautils")
+
+# Cache of done transitions per project, saves looking this up for
+# every issue
+transitions = {}
+
+# These words are not indexed by jira. Adding a * to them will cause
+# zero results to be returned.
+# From: https://confluence.atlassian.com/jiracorecloud/
+#       search-syntax-for-text-fields-765593720.html
+JQL_RESERVED_WORDS = [
+    "a",
+    "and",
+    "are",
+    "as",
+    "at",
+    "be",
+    "but",
+    "by",
+    "for",
+    "if",
+    "in",
+    "into",
+    "is",
+    "it",
+    "no",
+    "not",
+    "of",
+    "on",
+    "or",
+    "s",
+    "such",
+    "t",
+    "that",
+    "the",
+    "their",
+    "then",
+    "there",
+    "these",
+    "they",
+    "this",
+    "to",
+    "was",
+    "will",
+    "with"
+]
+
+
+def translate_issue_query(input_query, project):
+    """Create JQL query from summary match string
+
+    This function will append star to every word, and remove JQL_RESERVED_WORDS
+    and numeral only terms as they have no effect on the query anyway.
+    Then a query is constructed from the project string and
+    synthesised summary"""
```

Comment:
This is a good point, the current method will fail, when other jobs match all the terms in a job name. @mattt416 did suggest labels, but I wanted to avoid it so that we don't create many labels. So we could switch over to labels to avoid ambiguity, but it sounds like you aren't really in favour of this at all.

Reply:
Aside from the technical review here, this is a change that has been requested by all teams who have periodic jobs, as well as the AST team who are helping with triage. I don't believe it's for us to deny a request for functionality if that's how a team desires to work. If a team wishes to have issue-per-failure then they can always turn this off.

@chundu was particularly interested to see this functionality enabled